opentelemetry/src/logs/record.rs

```diff
@@ -26,7 +26,11 @@ pub trait LogRecord {
     fn set_body(&mut self, body: AnyValue);
 
     /// Adds multiple attributes.
-    fn add_attributes(&mut self, attributes: Vec<(Key, AnyValue)>);
+    fn add_attributes<I, K, V>(&mut self, attributes: I)
+    where
+        I: IntoIterator<Item = (K, V)>,
+        K: Into<Key>,
+        V: Into<AnyValue>;
```

Comment:
Not adding the default implementation to iteratively call `add_attribute` in API, let SDK decide how to add the attributes.